models/object_detection_nanodet/demo.py

```diff
@@ -0,0 +1,110 @@
+    parser.add_argument('--input_type', type=str, default='image', help="Input types: image or video")
+    parser.add_argument('--image_path', type=str, default='test_data/test1.jpg', help="image path")
+    parser.add_argument('--video_path', type=str, default='test_data/TestV1.mp4', help="video path")
+    parser.add_argument('--confidence', default=0.35, type=float, help='class confidence')
+    parser.add_argument('--nms', default=0.6, type=float, help='enter nms IOU threshold')
+    args = parser.parse_args()
+
+    if (args.input_type=="image"):
+        image = cv2.imread(args.image_path)
+        image = cv2. cvtColor(image, cv2.COLOR_BGR2RGB)
+        model_net = NanoDet(prob_threshold=args.confidence, iou_threshold=args.nms)
+
+        srcimg = image
+        drawimg = srcimg.copy()
+        a = time.time()
+        #image = model_net.infer(image)
+        result, left, top, ratioh, ratiow, det_bboxes, det_conf, det_classid = model_net.infer(image)
+        b = time.time()
+        print('Inference_Time:'+str(b-a)+' secs')
+        for i in range(det_bboxes.shape[0]):
+            xmin, ymin, xmax, ymax = max(int((det_bboxes[i,0] - left) * ratiow), 0), max(int((det_bboxes[i,1] - top) * ratioh), 0), min(
+                int((det_bboxes[i,2] - left) * ratiow), srcimg.shape[1]), min(int((det_bboxes[i,3] - top) * ratioh), srcimg.shape[0])
+            frame = drawPred(drawimg, det_classid[i], det_conf[i], xmin, ymin, xmax, ymax)
+
+        plt.imshow(frame)
+        plt.axis('off')
+        plt.show()
+
+    else:
+        model_net = NanoDet(prob_threshold=args.confidence, iou_threshold=args.nms)
+        colors = [(255, 255, 0), (0, 255, 0), (0, 255, 255), (255, 0, 0)]
+        capture = cv2.VideoCapture(args.video_path)
+        start = time.time_ns()
+
+        frame_count = 0
+        total_frames = 0
+        fps = -1
+
+        model_net = NanoDet(prob_threshold=args.confidence, iou_threshold=args.nms)
+
+        while True:
+            hasframe, frame = capture.read()
+            if not hasframe:
+                print("End of stream")
+                break
+
+            #image = model_net.infer(frame)
+            srcimg = frame
+            drawimg = srcimg.copy()
+            result, left, top, ratioh, ratiow, det_bboxes, det_conf, det_classid = model_net.infer(frame)
+            for i in range(det_bboxes.shape[0]):
+                xmin, ymin, xmax, ymax = max(int((det_bboxes[i,0] - left) * ratiow), 0), max(int((det_bboxes[i,1] - top) * ratioh), 0), min(
+                    int((det_bboxes[i,2] - left) * ratiow), srcimg.shape[1]), min(int((det_bboxes[i,3] - top) * ratioh), srcimg.shape[0])
+                image = drawPred(drawimg, det_classid[i], det_conf[i], xmin, ymin, xmax, ymax)
+
+            frame_count += 1
+            total_frames += 1
+
+            if frame_count >= 50:
+                end = time.time_ns()
+                fps = 1000000000 * frame_count / (end - start)
```

Comment:
You do not have to calculate FPS manually. Use `cv2.TickMeter()` to measure time, and you can get FPS by calling `getFPS()` as done below:
https://github.com/opencv/opencv_zoo/blob/54190541bb9f717144b255358ca6b37e3e47bf23/models/face_detection_yunet/demo.py#L118-L131

Reply:
reviewed changes, 
- used tm.getFPS()